docs/12.EXT-REFERENCE-MODULE.md

```diff
@@ -0,0 +1,177 @@
+#  Module API
+
+A JerryScript module is defined by calling the macro `JERRYX_MODULE` and supplying it with the name of the module as
+its first parameter and the name of an initialization function as its second parameter.
+
+**Note**: The `JERRYX_MODULE` macro depends on compiler support because it uses `__attribute__` extensions provided
+only by GCC and LLVM/clang to ensure that JerryScript's own module resolver (`jerryx_module_native_resolver`) will
+find a module that was declared with `JERRYX_MODULE`. If such compiler support is absent, currently
+`jerryx_module_native_resolver` will fail to find a module declared with `JERRYX_MODULE`.
```

Comment:
To tell you the truth I don't really like the compiles but does not work at runtime approach. That is awful to debug. I would prefer a compiler error to raise the attention. What I would prefer: conditional definition of macros / structs / functions which related to native modules, and all of these types should have `native` in their name to make their role clear.